src/lib/datepicker/datepicker.ts

```diff
@@ -526,3 +518,45 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
     }
   }
 }
+
+// TODO(mmalerba): We use a component instead of a directive here so the user can use implicit
+// template reference variables (e.g. #d vs #d="matDatepicker"). We can change this to a directive
+// if angular adds support for `exportAs: '$implicit'` on directives.
+/**
+ * Component responsible for managing the datepicker popup/dialog.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'mat-datepicker',
+  template: '',
+  exportAs: 'matDatepicker',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  providers: [{provide: MatDateSelectionModel, useClass: MatSingleDateSelectionModel}]
+})
+export class MatDatepicker<D> extends MatDatepickerBase<D> {}
```

Comment:
Currently the Datepicker inherits from the base as is. If this stays this way, we should simply make the the base class the `MatDatepicker<D>` and have `MatDaterangePicker<D>` derive from that. Making a base class was anticipatory.

Reply:
I like separating it, because I think we may find that it's useful for the subclasses to know which kind of selection model they're working with, while the base class shouldn't.